internal/lrucache.js

```diff
@@ -21,9 +21,9 @@ class LRUCache {
   }
 
   set (key, value) {
-    const deleted = this.delete(key)
+    this.delete(key)
 
```

Comment:
```suggestion

```

Reply:
This pull request was not meant as an optimization, but a fix to the LRU semantics. The two added unit tests for the lrucache.js module test the behavior.

Removing the `this.delete(key)` would make the LRU cache implementation into a mixture of LRU and FIFO caches. It may not be relevant with how the module is _currently_ used, but calling it an LRU cache could be an issue if someone were to expect it to work like a typical one in the future.

As such I recommend either keeping the delete call, removing it and calling the cache implementation something else than an LRU, or just closing this pull request. From the performance perspective removing or keeping the delete call seems to make no difference in the benchmarks.